sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_storage.py

```diff
@@ -187,6 +201,41 @@ def put(self, data, lease_period=None):
             lease_period = self._lease_period
         return blob.put(data, lease_period=lease_period)
 
+    def _check_and_set_folder_permissions(self):
+        try:
+            # Create path if it doesn't exist
+            os.makedirs(self._path, exist_ok=True)
+            # Windows
+            if os.name == "nt":
+                user = self._get_current_user()
+                if not user:
+                    logger.error(
+                        "Failed to retrieve current user. Skipping folder permission setup."
+                    )
+                    return False
+                result = subprocess.run(
```

Comment:
So we instantiate a `LocalFileStorage` instance for every exporter that is created. We do not want to be running a subprocess everytime this happens so we should probably add a check to see if permissions are already set before doing this.

Reply:
Additionally, if we want to support this only on our officially supported distro, it might be beneficial to implement all of this permission creation on the distro level, and have the exporter only check if permissions exist.